### DISCUSSION TITLE
Custom Fields handling in the Backend while Editing Content

### DISCUSSION THREAD

**Discussion by olivdee** [joomla/joomla-cms#39991] [2. Cold Features]
Is it in generall possible to implement (some day) the following 2 ideas into Joomla, or would it "break" current structure/logic/workflow?

1. The possibility of **picking a custom field while editing** an article? (NOW: you define a Custom Field and/or Group and pre-assign it, no way you can select another assigned field while editing the content)

2.  A feature for **re-arranging/dragging the custom fields while editing** an article.
 
3. Similar to (2): A feature for **re-arranging/dragging TinyMCE templates** (and other elements - mayber by manually defining in settings something like "Apply only on 1st Level div/p etc."

Maybe this is already somehow possible with an extension/plugin etc.??? So far I could not find one though...

For 1 & 2 see this animated GIF (looping) as an example (yes, this is not Joomla):

![Move-Add-Custom-Fields-while-editing](https://user-images.githubusercontent.com/2912892/222759853-ac9ac32e-326e-47dd-88d4-0454c4b0a94e.gif)

**Comment by Fedik** [comment]
I will link some existing similar topics
- https://github.com/joomla/joomla-cms/issues/36710
- https://github.com/joomla/joomla-cms/discussions/38639
- https://github.com/joomla/joomla-cms/discussions/19150

**Reply by olivdee** [reply]
Thank you very much! That was very insightful! Did spend the weekend reading all of that and all mentioned other sources in there. Puh, seems to be a difficult birth (which I understand). 

Our company is building Websites for customers since 2005 and using Joomla from day 1. For more complex projects we used CCK Extensions like K2, Seblod, Fabrik. Later we were also using Drupal, Typo3, ProcessWire - depending on the project and available developer. Of course we also touched Wordpress with some projects as well - which I say it is the worse. 

All in all I believe, that if Joomla would add a simple CCK functionality as discussed on the links, it would gain a lot of traction. Many projects would be possible to be completed in Joomla without a heavy 3rd party extension. I loved Joomla from day 1, and still love it and use it whenever possible - it would be nice to use it for ALL projects coming in the future though. 

Bigger projects tend to be content heavy/complex and with many external content editors - therefore a "content type" similar feature makes great sense. I will try to optimize **backend/content creation workflow for editors** in new projects by using Joomla via "**content admin menu filters**" as explained in https://github.com/joomla/joomla-cms/issues/20233 and https://magazine.joomla.org/all-issues/february-2022/joomla-4-for-clients-part-2-creating-a-custom-admin-experience, as well as **subforms** and the soon coming **d2 Profiles** (https://data2site.com/extensions/d2-profiles). But I believe, one big part missing will be the possibility to re-arrange those "fake" content types or custom fields while creating/editing the content. But maybe I am wrong.
